src/main/java/seedu/guest/logic/parser/AddCommandParser.java

```diff
@@ -36,26 +38,43 @@ public class AddCommandParser implements Parser<AddCommand> {
     public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
-                        PREFIX_ROOM, PREFIX_DATE_RANGE, PREFIX_NUMBER_OF_GUESTS, PREFIX_IS_ROOM_CLEAN);
+                        PREFIX_ROOM, PREFIX_DATE_RANGE, PREFIX_NUMBER_OF_GUESTS, PREFIX_IS_ROOM_CLEAN, PREFIX_REQUEST);
 
         if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ROOM,
                 PREFIX_DATE_RANGE, PREFIX_NUMBER_OF_GUESTS, PREFIX_IS_ROOM_CLEAN)
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
 
-        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
-        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
-        Room room = ParserUtil.parseRoom(argMultimap.getValue(PREFIX_ROOM).get());
-        DateRange dateRange = ParserUtil.parseDateRange(argMultimap.getValue(PREFIX_DATE_RANGE).get());
-        NumberOfGuests numberOfGuests = ParserUtil
-                .parseNumberOfGuests(argMultimap.getValue(PREFIX_NUMBER_OF_GUESTS).get());
-        IsRoomClean isRoomClean = ParserUtil
-                .parseIsRoomClean(argMultimap.getValue(PREFIX_IS_ROOM_CLEAN).get());
-        Bill bill = new Bill();
-        Guest guest = new Guest(name, phone, email, room, dateRange, numberOfGuests, isRoomClean, bill);
-        return new AddCommand(guest);
+        if (arePrefixesPresent(argMultimap, PREFIX_REQUEST)) {
+            Request request = ParserUtil.parseRequest(argMultimap.getValue(PREFIX_REQUEST).get());
```

Comment:
Should this line be after the Bill line for consistency?

Reply:
changed